corgi/collectors/brew.py

```diff
@@ -206,6 +219,10 @@ def _extract_bundled_provides(
             if not component:
                 continue
             bundled_component = cls._check_maven_component(component, version)
+            if bundled_component:
+                bundled_components.append(bundled_component)
+                continue
+            bundled_component = cls._check_npm_component(component, version)
```

Comment:
You're right about the situation where after splitting on dash, we end up with `component_type == "rh"` and `component == "nodejs12-nodejs-zlib"`. The issue I encountered trying to solve that is that there are many components with similarly structured names but perhaps different semantics, for example:

```
['rh-eclipse-guice-assistedinject', 'rh-nodejs6-nodejs-preserve', 'rh-eclipse-ed25519-java', 'rh-eclipse46-eclipse-oprofile', 'rh-postgresql12-container', 'rh-ror41-rubygem-rack', 'rh-java-common-netty', 'rh-postgresql10-postgresql-plpython', 'rh-eclipse-xmlgraphics-commons', 'rh-php73-runtime', 'rh-maven35-rhino', 'rh-maven35-osgi-compendium', 'rh-git218-git-instaweb', 'rh-eclipse46-eclipse-mylyn', 'rh-postgresql96-postgresql-static', 'rh-nodejs8-nodejs-yargs', 'rh-php71-php-mbstring', 'rh-nodejs6-nodejs-realize-package-specifier', 'rh-perl526-perl-Compress-Bzip2', 'rh-ruby24-ruby-libs']
```

It seems like these follow the general pattern of `rh-{framework}-{package}`, but even the randomly selected subset above illustrates the problems with trying to handle these generally. Some specify a specific framework/bundle release (`rh-eclipse46-eclipse-oprofile`) while some don't (`rh-eclipse-xmlgraphics-commons`).

My first inclination would be to just lstrip "rh-", since I don't think that'll ever be a meaningful component type, but the meaning of `3` on [line 231](https://github.com/RedHatProductSecurity/component-registry/pull/557/files#diff-35566a508ba49d2e035c1d2f80337b84909bf7628df7e9082563766cd96b3927R231) is not immediately clear to me, and I wouldn't want to break whatever case that's handling.

Reply:
I suspect the meaning of 3 of line 231 is just that we aren't going to try and detect the type where there are more than 2 dashes. If you think you can do a better job of detecting the type by lstrip- "rh-" by all means, give it a go.

Reply:
In this latest change, I've kept the separate handling for each component type because without doing so, we lose information. For example, given a bundled component named `bundled(rh-nodejs14-icu)`, if we strip the leading `rh-` and then split on `-`, end up with a component_type of `nodejs14` and a component of `icu`. We can determine that this is an NPM package by matching `nodejs*` in the component_type, but then we lose any distinction between say `nodejs14-icu` and `nodejs16-icu`, because the component_type is split.

For that reason, I think it's better to detect these cases before splitting the component into component_type and component, but please let me know if there's anything I'm missing about that approach.